harness/determined/cli/user.py

```diff
@@ -218,6 +243,110 @@ def edit(args: argparse.Namespace) -> None:
         raise errors.CliError("No field provided. Use 'det user edit -h' for usage.")
 
 
+def render_token_info(token_info: Sequence[bindings.v1TokenInfo]) -> None:
+    values = []
+    for ti in token_info:
+        value = [
+            ti.id,
+            ti.userId,
+            ti.description,
+            ti.createdAt,
+            ti.expiry,
+            ti.revoked,
+            ti.tokenType,
+        ]
+        values.append(value)
+    render.tabulate_or_csv(TOKEN_HEADERS, values, False)
+
+
+def print_token_info(data: Sequence[bindings.v1TokenInfo], args: argparse.Namespace) -> None:
+    if data:
+        json_data = [elem.to_json() for elem in data]
+        if len(data) == 1:
+            json_data = json_data[0]
+    else:
+        json_data = {}
```

Comment:
It looks like this is only called when people call `det token list --json` or `--yaml`.

That means people are probably intending to consume the output in a script.

That means that you should be consistent about the data format that you emit.  Right now:
 - if there are no tokens, you emit an empty dict
 - if there is one token, you emit a non-empty dict
 - if there are multiple tokens, you emit a list of dicts

Instead, just always emit a list of dicts, which may be length 0, or 1, or >1.

Also, you probably don't need a helper function anymore with that change.

```python
if args.json or args.yaml:
   json_data = [t.to_json() for t in resp.tokenInfo]
   if args.json:
       render.print_json(json_data)
   else:
       print(util.yaml_safe_dump(json_data, default_flow_style=False))
else:
    render_token_info(resp.tokenInfo)
```

Reply:
Resolving it as it is now implemented